dca/tl_nc_queue.php

```diff
@@ -8,6 +8,8 @@
  * @license    LGPL
  */
 
+\Contao\System::loadLanguageFile('tl_nc_queue');
```

Comment:
Is this the right solution for the missing translation for `re-queueConfirmation` while `cache:warmup`?

Reply:
That does not look right to me, that's the reason for this?

Reply:
while `cache:warmup` there is a warning for a missing translation array key `re-queueConfirmation`

Reply:
Are there any better solutions?

Reply:
In the core we added a null check to the usage. Like this:

```php
if(!confirm(\'' . $GLOBALS['TL_LANG']['tl_nc_queue']['re-queueConfirmation'] ?? null . '\')) […]
```

Reply:
Possibly related to https://github.com/contao/contao/pull/2536 ?

The line should not be necessary.

Reply:
But if I do not load the language file I get the following error while `cache:warmup`: `PHP Warning:  Undefined array key "re-queueConfirmation"`

Reply:
Hm, I could not reproduce. I installed current `dev-main` with PHP `8.0.9` and there were no errors during `cache:warmup`.

Reply:
@rabauss can you try to reproduce this again and post the complete stack trace?

Reply:
For me the Language file of `tl_nc_queue` is really not loaded while `cache:warmup` - here is the callstack:

```
tl_nc_queue.php:17, include()
DcaLoader.php:93, Contao\DcaLoader->loadDcaFiles()
DcaLoader.php:70, Contao\DcaLoader->load()
Controller.php:1360, Contao\Controller::loadDataContainer()
DcaExtractor.php:380, Contao\DcaExtractor->createExtract()
DcaExtractor.php:128, Contao\DcaExtractor->__construct()
DcaExtractor.php:150, Contao\DcaExtractor::getInstance()
ContaoCacheWarmer.php:197, Contao\CoreBundle\Cache\ContaoCacheWarmer->generateDcaExtracts()
ContaoCacheWarmer.php:94, Contao\CoreBundle\Cache\ContaoCacheWarmer->warmUp()
CacheWarmerAggregate.php:98, Symfony\Component\HttpKernel\CacheWarmer\CacheWarmerAggregate->warmUp()
CacheWarmupCommand.php:81, Symfony\Bundle\FrameworkBundle\Command\CacheWarmupCommand->execute()
Command.php:256, Symfony\Bundle\FrameworkBundle\Command\CacheWarmupCommand->run()
Application.php:989, Contao\ManagerBundle\Console\ContaoApplication->doRunCommand()
Application.php:96, Contao\ManagerBundle\Console\ContaoApplication->doRunCommand()
Application.php:290, Contao\ManagerBundle\Console\ContaoApplication->doRun()
Application.php:82, Contao\ManagerBundle\Console\ContaoApplication->doRun()
Application.php:166, Contao\ManagerBundle\Console\ContaoApplication->run()
contao-console:37, {main}()
```

Reply:
Is that the stack trace for the warning?

Reply:
Oh crazy, now I had some trouble with my docker containers and I had to rebuild my project with php 8.0.10 the warning does not occur anymore without the line `\Contao\System::loadLanguageFile('tl_nc_queue')`

Reply:
I still cannot reproduce with PHP 8 and Contao 4.9. You should never have to load the language file of the same DCA yourself. You only need to load language files, if your DCA accesses translations of different DCAs/language domains.

Reply:
I can reproduce it only with an existing database!

Reply:
Well yes I am also testing with an existing database :).

Reply:
Thanks @fritzmg  for the support - for me the js-warning for `re-queueConfirmation` is also working with the `??` check!